Automattic/wp-calypso
# Make site public directly from <EligibilityWarnings>

#### Changes proposed in this Pull Request

If a site is private or unlaunched, and doesn't need a plan upgrade, then clicking the "Continue" button should launch/un-private the site automatically.

This is part of #38141 

I've made it so that if a site is blocked from going Atomic because it's private, then clicking "Continue" will change the visibility setting to "Hidden" (not indexed), not "Public". Given they probably weren't in the mindset of making their site public, it seems like changing to "hidden" is the smallest change to the settings that will allow them to go Atomic.

This PR doesn't update the copy to make it clear whether a site is private or unlaunched, that's #38324 

* `<EligibilityWarnings>` component can automatically launch sites
* `<EligibilityWarnings>` component can automatically set site visibility to "hidden"
* `launchSite` selector returns a promise that resolves to `true` if the site was launched
* Changed unrelated `location.href` to `window.location.href` to silence eslint so I could commit

#### Testing instructions



* Create an unlaunched site with a business plan
* View eligibility warnings by uploading plugin or theme, or change hosting settings
* The continue button should launch your site and proceed to the next page (e.g. plugin upload page)
* Create a site with a business plan, launch it, then set visibility settings to private
* View eligibility warnings by uploading plugin or theme, or change hosting settings
* The continue button should change the visibility setting to "hidden" and proceed to the next page (e.g. plugin upload page)
